component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java

```diff
@@ -70,15 +79,46 @@ public void testPoint() throws Exception {
             public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
                 for (Event event : inEvents) {
                     logger.info(event);
-                    Boolean isWithin = (Boolean) event.getData(0);
-                    AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                    count.incrementAndGet();
+                    if (count.get() == 1) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+
+                    }
+                    if (count.get() == 2) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 3) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 4) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 5) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 6) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
                 }
             }
         });
         siddhiAppRuntime.start();
         generateEvents(siddhiAppRuntime);
-        Thread.sleep(1000);
-        AssertJUnit.assertEquals(expectedResult.size(), eventCount);
+        SiddhiTestHelper.waitForEvents(100, 6, count, 60000);
+        AssertJUnit.assertEquals(6, count.get());
```

Comment:
Don't we need to stop the siddhiAppRuntime ?

Reply:
Fixed by 26e6366